release-notes/7.0/preview/api-diff/preview6/Microsoft.NETCore.App/7.0-preview6_System.IO.md

```diff
@@ -0,0 +1,43 @@
+# System.IO
+
+``` diff
+ namespace System.IO {
+     public static class Directory {
++        [UnsupportedOSPlatformAttribute("windows")]
++        public static DirectoryInfo CreateDirectory(string path, UnixFileMode unixCreateMode);
+     }
+     public static class File {
++        [UnsupportedOSPlatformAttribute("windows")]
++        public static UnixFileMode GetUnixFileMode(SafeFileHandle fileHandle);
++        [UnsupportedOSPlatformAttribute("windows")]
++        public static UnixFileMode GetUnixFileMode(string path);
++        [UnsupportedOSPlatformAttribute("windows")]
++        public static void SetUnixFileMode(SafeFileHandle fileHandle, UnixFileMode mode);
++        [UnsupportedOSPlatformAttribute("windows")]
++        public static void SetUnixFileMode(string path, UnixFileMode mode);
+     }
+     public sealed class FileStreamOptions {
++        public UnixFileMode? UnixCreateMode { get; [UnsupportedOSPlatformAttribute("windows")] set; }
+     }
+     public abstract class FileSystemInfo : MarshalByRefObject, ISerializable {
++        public UnixFileMode UnixFileMode { get; [UnsupportedOSPlatformAttribute("windows")] set; }
+     }
++    [FlagsAttribute]
++    public enum UnixFileMode {
++        GroupExecute = 8,
++        GroupRead = 32,
++        GroupWrite = 16,
++        None = 0,
++        OtherExecute = 1,
++        OtherRead = 4,
++        OtherWrite = 2,
++        SetGroup = 1024,
++        SetUser = 2048,
++        StickyBit = 512,
++        UserExecute = 64,
++        UserRead = 256,
++        UserWrite = 128,
```

Comment:
nit: this is ordered by name, but the [ref ](https://github.com/dotnet/runtime/pull/69980/files#diff-cec8e6f471b4193246bdc0107b0dd7cbe131fb7fd189b288b37269c333d1171dR10090-R10106) orders by value.